pinot-core/src/main/java/com/linkedin/pinot/core/realtime/converter/stats/RealtimeColumnStatistics.java

```diff
@@ -146,11 +161,42 @@ public boolean hasNull() {
 
   @Override
   public PartitionFunction getPartitionFunction() {
-    return null;
+    return partitionFunction;
   }
 
   @Override
   public List<IntRange> getPartitionRanges() {
-    return null;
+    return partitionRanges;
   }
+
+  /**
+   * For each element in the dictionary for a column
+   * Checks if the given value lies within one of the partition ranges. If the value is not within
+   * any of the partition ranges, partitioning is dropped. The effect of that is:
+   * <ul>
+   *   <li> Subsequent values will not be checked against partition ranges. </li>
+   *   <li> Partition information will not be written out to the metadata for this column. </li>
+   * </ul>
+   *
+   */
+  void checkPartition() {
+    // Iterate over the dictionary to check the partitioning
+    final int length = _dictionaryReader.length();
+    for (int i = 0; ((i < length) && (partitionFunction != null)); i++) {
+      int partition = partitionFunction.getPartition(_dictionaryReader.get(i));
+      for (IntRange partitionValue : partitionRanges) {
+        if (partition < partitionValue.getMinimumInteger() || partition > partitionValue.getMaximumInteger()) {
```

Comment:
IntRange has a `containsInteger` method, this would be more readable as `!partitionValue.containsInteger(partition)` (especially if you rename partitionValue to partitionRange).